app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -79,11 +79,11 @@ def url_contains_version
     end
 
     def allow_external_network
-      Foreman::Cast.to_bool(attrs[:allow_external_network])
```

Comment:
Sorry for miscommunication, I thought that before we used here `to_bool` from the `app/models/compute_resource.rb`, not the `Foreman::Cast.to_bool`.

Here we should indeed replace the behavior with `ActiveRecord::Type::Boolean.new.deserialize`,
sorry for confusion